packages/babel-plugin-transform-es2015-classes/src/vanilla.js

```diff
@@ -25,15 +25,20 @@ const noMethodVisitor = {
 };
 
 const verifyConstructorVisitor = visitors.merge([noMethodVisitor, {
-  Super(path) {
+  Super(path, state) {
     if (
       this.isDerived && !this.hasBareSuper &&
-      !path.parentPath.isCallExpression({ callee: path.node })
+      !path.parentPath.isCallExpression({ callee: path.node }) &&
+      !state.inArrowFunctionExpression
     ) {
       throw path.buildCodeFrameError("'super.*' is not allowed before super()");
     }
   },
 
+  ArrowFunctionExpression(path, state) {
+    state.inArrowFunctionExpression = true;
```

Comment:
Yes, thanks a lot @loganfsmyth. that's right, I will update it.

I thinking how we should document better the state argument.